### DISCUSSION TITLE
Where to find the damn GUID of Windows Runtime API COM Interface?

### DISCUSSION THREAD

**Discussion by GF-Huang** [microsoft/WindowsAppSDK#4193] [Q&A]
For this interface: https://learn.microsoft.com/en-us/windows/win32/api/windows.graphics.capture.interop/nn-windows-graphics-capture-interop-igraphicscaptureiteminterop and https://learn.microsoft.com/en-us/uwp/api/windows.graphics.capture.graphicscaptureitem?view=winrt-22621

I know there's a project use it:
https://github.com/robmikh/ManagedScreenshotDemo/blob/9a47612e1c3a9f5f70d142c588a23111930eaddb/Robmikh.WindowsRuntimeHelpers/CaptureHelper.cs#L11

But where to find the `79C3F95B-31F7-4EC2-A464-632EF5D30760`?

**Comment by DarranRowe** [comment]
![Screenshot 2024-02-12 235648](https://github.com/microsoft/WindowsAppSDK/assets/52577874/37c4c246-74b5-456c-a856-5e07aaf565e6)

```
//---------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
//---------------------------------------------------------------------------

#if defined(_MSC_VER) && (_MSC_VER >= 1020)
#pragma once
#endif

#include <windows.ui.composition.h>
#include <windows.graphics.capture.h>
#include <sdkddkver.h>

#undef INTERFACE
#define INTERFACE IGraphicsCaptureItemInterop
DECLARE_INTERFACE_IID_(IGraphicsCaptureItemInterop, IUnknown, "3628E81B-3CAC-4C60-B7F4-23CE0E0C3356")
{
    IFACEMETHOD(CreateForWindow)(
        HWND window,
        REFIID riid,
        _COM_Outptr_ void ** result
        ) PURE;

    IFACEMETHOD(CreateForMonitor)(
        HMONITOR monitor,
        REFIID riid,
        _COM_Outptr_ void ** result
        ) PURE;

};
```

There is no CLSID attached to the GraphicsCaptureItem runtime class. WinRT runtime classes are activated using the fully qualified name, so it would use Windows.Graphics.Capture.GraphicsCaputreItem when passed to RoGetActivationFactory or RoActivateInstance. Higher level functionality, including the C# and C++/WinRT projections are built on this.

![Screenshot 2024-02-13 000147](https://github.com/microsoft/WindowsAppSDK/assets/52577874/4ee1e253-7b77-460f-8732-b45e6a488ed2)

```
            [contract(Windows.Foundation.UniversalApiContract, 6.0)]
            [exclusiveto(Windows.Graphics.Capture.GraphicsCaptureItem)]
            [uuid(79C3F95B-31F7-4EC2-A464-632EF5D30760)]
            interface IGraphicsCaptureItem : IInspectable
            {
                [propget] HRESULT DisplayName([out] [retval] HSTRING* value);
                [propget] HRESULT Size([out] [retval] Windows.Graphics.SizeInt32* value);
                [eventadd] HRESULT Closed([in] Windows.Foundation.TypedEventHandler<Windows.Graphics.Capture.GraphicsCaptureItem*, IInspectable*>* handler, [out] [retval] EventRegistrationToken* token);
                [eventremove] HRESULT Closed([in] EventRegistrationToken token);
            }
```
That IID that you listed is for the IGraphicsCaptureItem interface, which should be seen as the primary interface for GraphicsCaptureItem instances. Any static methods would be on the IGraphicsCaptureItemStatics and IGraphicsCaptureItemStatics2 interfaces.

**Comment by CaptainSpalding** [comment]
under 'shared' and 'um' in the SDK mostly,  also in regedit under 'HKEY_CLASSES_ROOT\CLSID' and 'HKEY_CLASSES_ROOT\Interface'. its tedious they should have a tool. I may make one who knows

**Reply by GF-Huang** [reply]
Microslop Bugrows 11 is a bullshit, there’s no point wasting time on it.
